### DISCUSSION TITLE
Timelapse with -skip_frame nokey is choppy

### DISCUSSION THREAD

**Discussion by LukeHandle** [blakeblackshear/frigate#14126] [General Support] [answered]
With the default configuration for the timelapse (`DEFAULT_TIME_LAPSE_FFMPEG_ARGS = "-vf setpts=0.04*PTS -r 30"`), I found this was kinda choppy, I'd get the same original frame shown for a few frames of exported timelapse, followed by a large-ish jump forward in time.

I was hoping for a smoother playback / each frame shown for less time. I tried increasing the framerate to 60, but had the same sort of outcome.

I tested manually a little with shorter clips (exec'd into container) and discovered it's the `TIMELAPSE_DATA_INPUT_ARGS = "-an -skip_frame nokey"` that is causing the choppiness. When removing this, I get the sort of sped-up / video look of the timelapse (vs. the slideshow).

It seems this option greatly improves the performance of the export as it doesn't need to read the full video, and I agree this option is much like a traditional timelapse with a photo every X seconds.

I did a normal export and then re-encoded with the following, it took ~15 minutes from 3h20m @ 15fps -> 6m @ 60fps. ffmpeg reported a speed of 0.4 (24fps).

```
ffmpeg -hide_banner -hwaccel cuda -hwaccel_output_format cuda -extra_hw_frames 8 -an -y -i /media/frigate/exports/side_bfq2ko.mp4 -c:v hevc_nvenc -vf 'setpts=0.03*PTS' -r 60 -movflags +faststart /media/frigate/exports/side_bfq2ko_timelapse.mp4
```

===

Does it make sense to have some customizability here, or is there a better way I'm missing to achieve the same sort of sped-up effect?

_Ahh, I just read number 7 on https://github.com/blakeblackshear/frigate/discussions/12724 - I guess there might not be a huge appetite here_

For clarity, my aim was recording building work and wanting to quick summary of interesting days which I can keep longer-term.

**Comment by NickM-27** [comment, marked as answer]
the use cases for timelapses vary widely between users, like you saw we don't want to build out support for the time lapse feature to support all of these different use cases. I'd suggest a script to run ffmpeg along with using the recordings API would solve this usecase

**Reply by LukeHandle** [reply]
It seems you are clairvoyant :D : https://github.com/blakeblackshear/frigate/commit/94fd75e014e2ef66daf78b9d92ecdef7eb7242ea#diff-a6a41989fc49a96d8b12e053eee31df7029d7ac84619941cdf348aa4e2576a10R98

With that change, yea, this should be fairly scriptable. Thanks!
